grafana/data_source_dashboards.go

```diff
@@ -0,0 +1,103 @@
+package grafana
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func DatasourceDashboards() *schema.Resource {
+	return &schema.Resource{
+		Description: `
+Datasource for retrieving all dashboards. Specify list of folder IDs to search in for dashboards.
+
+* [Official documentation](https://grafana.com/docs/grafana/latest/dashboards/)
+* [Folder/Dashboard Search HTTP API](https://grafana.com/docs/grafana/latest/http_api/folder_dashboard_search/)
+* [Dashboard HTTP API](https://grafana.com/docs/grafana/latest/http_api/dashboard/)
+`,
+		ReadContext: dataSourceReadDashboards,
+		Schema: map[string]*schema.Schema{
+			"folder_ids": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "Numerical IDs of Grafana folders containing dashboards. Specify to filter for dashboards by folder (eg. `[0]` for General folder), or leave blank to get all dashboards in all folders.",
+				Elem:        &schema.Schema{Type: schema.TypeInt},
+			},
+			"tags": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "List of string Grafana dashboard tags to search for, eg. `[\"prod\"]`. Used only as search input, i.e., attribute value will remain unchanged.",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"dashboards": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"title": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"uid": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"folder_title": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceReadDashboards(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*client).gapi
+	var diags diag.Diagnostics
+	params := url.Values{
+		"limit": {"5000"},
```

Comment:
I'd make this a schema attribute with a 5000 default. Could be useful to find the first matching dashboard for example